packages/dart_frog/lib/src/context.dart

```diff
@@ -43,6 +43,21 @@ This can happen if $T was not provided to the request context:
     return (value as T Function())();
   }
 
+  /// Attempt to lookup an instance of [T] from the [request] context.
+  ///
+  /// Returns `null` if [T] is not available within the provided
+  /// [request] context.
+  T? tryRead<T>() {
```

Comment:
Nit: If we end up adding `tryRead` I think we should modify the implementation to not catch the `StateError`, but instead implement `tryRead` so it does what `read` is doing right now but without the `null` check to avoid creating an additional `StateError` object. We could then base the `read` implementation on `tryRead` for code sharing.